src/compiler/transformers/jsx.ts

```diff
@@ -348,7 +348,10 @@ namespace ts {
             return element;
         }
 
-        function transformJsxSpreadAttributeToSpreadAssignment(node: JsxSpreadAttribute) {
+        function transformJsxSpreadAttributeToProps(node: JsxSpreadAttribute) {
+            if (isObjectLiteralExpression(node.expression)) {
+                return node.expression.properties;

```

Comment:
This isn't like 100% correct when it comes to JS semantics, but TS never was 100% correct anyway (eg. TS doesn't emit 100% spec-compliant output for a case like [this](https://github.com/babel/babel/pull/11471)).

Reply:
What isn't correct about it?

Reply:
I think this should be OK for the "sane code" but probably once you start thinking about malicious getters, setters, and stuff then you could figure out some edge case that would change behavior with this change. But then it means that the code compiled with TS doesn't work exactly the same as the code compiled with Babel because Babel applies this optimization. 

As mentioned though... TS isn't 100% correct even today so I'm not sure if a super subtle (and only for weird code) behavior change changes the "final balance" all that much.

TS is using `Object.assign` semantics for spreads (and that calls setters on the target object), whereas the correct semantic is to define properties (and that doesn't call setters). There are also cases like in the referenced Babel PR:
```js
var p = { a: 1 }
var classicAssign = Object.assign({}, { get nefariousGetter() { (p = { evil: 'muahauhauha' }); return 42 } }, { p }) // {"nefariousGetter":42,"p":{"a":1}}
var correct = { ...{ get nefariousGetter() { (p = { evil: 'muahauhauha' }); return 42 } }, p } // {"nefariousGetter":42,"p":{"evil":"muahauhauha"}}
```

😅